TargetRenamer/Trados.TargetRenamer/Trados.TargetRenamer/TargetRenamer.cs

```diff
@@ -1,27 +1,120 @@
             base.OnInitializeTask();
         }
+
         protected override void ConfigureConverter(ProjectFile projectFile, IMultiFileConverter multiFileConverter)
         {
-            //In here you should add your custom bilingual processor to the file converter
-            throw new NotImplementedException();
+	        var targetLanguage = projectFile.GetLanguageDirection().TargetLanguage.IsoAbbreviation.ToString();
+	        var file = projectFile.LocalFilePath;
+	        var fileName = Path.GetFileNameWithoutExtension(file);
+	        var filePath = Path.GetDirectoryName(file);
+	        var fileExtension = Path.GetExtension(file);
+	        string newFileName;
+
+	        if (_settings.AppendAsPrefix)
+	        {
+		        fileName = CreateFileNameWithPrefix(fileName, fileExtension, targetLanguage);
+	        }
+
+	        if (_settings.AppendAsSuffix)
+	        {
+		        fileName = CreateFileNameWithSuffix(fileName, fileExtension, targetLanguage);
+	        }
+
+	        if (_settings.UseRegularExpression)
+	        {
+		        fileName = CreateFileNameWithRegularExpression(fileName, fileExtension, targetLanguage);
+	        }
+
+	        if (_settings.UseCustomLocation)
+	        {
+		        newFileName = Path.Combine(_settings.CustomLocation, fileName);
+	        }
+	        else
+	        {
+				newFileName = Path.Combine(filePath, fileName);
+	        }
+
+	        if (_settings.OverwriteTargetFiles)
+	        {
+				File.Move(file, newFileName);
+	        }
+	        else
+	        {
+		        File.Copy(file, newFileName);
+	        }
+        }
+
+        private string CreateFileNameWithRegularExpression(string newFileName, string fileExtension, string targetLanguage)
+        {
+			// Maybe add an option to IgnoreCase?
+	        var newName = Regex.Replace(newFileName, _settings.RegularExpressionSearchFor,
+		        _settings.RegularExpressionReplaceWith) + fileExtension;
+	        return newName;
+        }
+
+        private string CreateFileNameWithSuffix(string newFileName, string fileExtension, string targetLanguage)
+        {
+			if (_settings.AppendTargetLanguage)
+			{
+				if (_settings.UseShortLocales)
+				{
+					newFileName = newFileName + _settings.Delimitator + targetLanguage.Substring(0, 2) + fileExtension;
+				}
+				else
+				{
+					newFileName = newFileName + _settings.Delimitator + targetLanguage.Substring(0,2) + fileExtension;
+				}
+			}
+			else if (_settings.AppendCustomString)
+			{
+				newFileName = newFileName + _settings.Delimitator + _settings.CustomString + fileExtension;
+			}
+
+			return newFileName;
+		}
+
+        private string CreateFileNameWithPrefix(string newFileName, string fileExtension, string targetLanguage)
+        {
+	        if (_settings.AppendTargetLanguage)
+	        {
+		        if (_settings.UseShortLocales)
+		        {
+			        newFileName = targetLanguage.Substring(0, 2) + _settings.Delimitator + newFileName +
```

Comment:
Should we consider using string builder for the name of the file?

Reply:
For a fixed and low (<5) number of strings it is faster to concatenate because the compiler might combine the operations into a single one